solidity/contracts/converter/ConverterBase.sol

```diff
@@ -474,6 +507,65 @@ contract ConverterBase is IConverter, TokenHandler, TokenHolder, ContractRegistr
 		return _targetAmount.mul(conversionFee).div(CONVERSION_FEE_RESOLUTION);
 	}
 
+	/**
+	 * @dev returns the protocol fee for a given target amount
+	 *
+	 * @param _targetAmount target amount
+	 *
+	 * @return calculated protocol fee
+	 */
+	function calculateProtocolFee(address _targetToken, uint256 _targetAmount) internal returns (uint256) {
+		uint256 _protocolFee = getProtocolFeeFromSwapSettings();
+		uint256 calculatedProtocolFee = _targetAmount.mul(_protocolFee).div(1e20);
+		protocolFeeTokensHeld[_targetToken] = protocolFeeTokensHeld[_targetToken].add(calculatedProtocolFee);
+		return calculatedProtocolFee;
+	}
+
+	function swapSettingsContractName() public constant returns(bytes32) {
+		return "SwapSettings";
+	}
+
+	/**
+	 * @dev get the protocol fee from the SovrynSwapNetwork using the registry.
+	 *
+	 * @return protocol fee.
+	 */
+	function getProtocolFeeFromSwapSettings() public view returns (uint256) {
+		address swapSettingsAddress = IContractRegistry(registry).addressOf(swapSettingsContractName());
+		return ISwapSettings(swapSettingsAddress).protocolFee();
+	}
+
+	/**
+	 * @dev get the wrbtc address from sovryn swap network
+	 *
+	 * @return wrbtc address
+	 */
+	function getWrbtcAddressFromSwapSettings() public view returns (address) {
+		address swapSettingsAddress = IContractRegistry(registry).addressOf(swapSettingsContractName());
+		return ISwapSettings(swapSettingsAddress).wrbtcAddress();
+	}
+
+	/**
+	 * @dev get the sov token address from sovryn swap network
+	 *
+	 * @return sov token address
+	 */
+	function getSOVTokenAddressFromSwapSettings() public view returns (address) {
+		address swapSettingsAddress = IContractRegistry(registry).addressOf(swapSettingsContractName());
+		return ISwapSettings(swapSettingsAddress).sovTokenAddress();
+	}
+
+	/**
+	 * @dev get the feesController address from sovryn swap network
+	 *
+	 * @return feesController address (protocol feeSharingProxy) 
```

Comment:
You could make all of these getters internal to save gas. This information is publicly available on the settings contract already.